stream_alert_cli/helpers.py

```diff
@@ -25,39 +26,109 @@
+            quiet (boolean): Whether to show command output or hide it
+
+    """
+    default_error_message = "An error occurred while running: {}".format(
+        ' '.join(runner_args)
+    )
+    error_message = kwargs.get('error_message', default_error_message)
 
-        default_cwd = 'terraform'
-        cwd = kwargs.get('cwd', default_cwd)
+    default_cwd = 'terraform'
+    cwd = kwargs.get('cwd', default_cwd)
 
-        stdout_option = None
-        if kwargs.get('quiet'):
-            stdout_option = open(os.devnull, 'w')
+    stdout_option = None
+    if kwargs.get('quiet'):
+        stdout_option = open(os.devnull, 'w')
 
-        try:
-            subprocess.check_call(runner_args, stdout=stdout_option, cwd=cwd)
-        except subprocess.CalledProcessError as e:
-            LOGGER_CLI.error('Return Code %s - %s', e.returncode, e.cmd)
-            return False
+    try:
+        subprocess.check_call(runner_args, stdout=stdout_option, cwd=cwd)
+    except subprocess.CalledProcessError as e:
+        LOGGER_CLI.error('Return Code %s - %s', e.returncode, e.cmd)
+        return False
+
+    return True
 
-        return True
+
+def format_lambda_test_record(test_record):
+    """Create a properly formatted Kinesis, S3, or SNS record.
+
+    Supports a dictionary or string based data record.  Reads in
+    event templates from the test/integration/templates folder.
+
+    Args:
+        test_record: Test record metadata dict with the following structure:
+            data - string or dict of the raw data
+            description - a string describing the test that is being performed
+            trigger - bool of if the record should produce an alert
+            source - which stream/s3 bucket originated the data
+            service - which aws service originated the data
+            compress (optional) - if the payload needs to be gzip compressed or not
+
+    Returns:
+        dict in the format of the specific service
+    """
+    service = test_record['service']
+    source = test_record['source']
+    compress = test_record.get('compress')
+
+    data_type = type(test_record['data'])
+    if data_type == dict:
+        data = json.dumps(test_record['data'])
+    elif data_type in (unicode, str):
+        data = test_record['data']
+    else:
+        LOGGER_CLI.info('Invalid data type: %s', type(test_record['data']))
+        return
+
+    # Get the template file for this particular service
+    template_path = os.path.join(DIR_TEMPLATES, '{}.json'.format(service))
+    with open(template_path, 'r') as service_template:
+        try:
+            template = json.load(service_template)
+        except ValueError as err:
+            LOGGER_CLI.error('Error loading %s.json: %s', service, err)
+            return
+
+    if service == 's3':
+        # Set the S3 object key to a random value for testing
+        test_record['key'] = ('{:032X}'.format(random.randrange(16**32)))
```

Comment:
I actually just checked to see why this passed integration tests - none of the `rules` we have in the integration tests are configured with `"service": "s3"`. There are also none from kinesis configured to use `compress`, so that's also why that's not failing.

We should either add more encompassing integration tests or we should (need to) add cli unit tests that would have caught this.